### PR TITLE
Fix BA events sending problems

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-86

### CODE REVIEW

File: internal/pkg/service/buffer/dependencies/mocked.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func (v *mocked) StatsCacheNode() *statistics.CacheNode {
 
 func (v *mocked) EventSender() *event.Sender {
 	if v.eventSender == nil {
-		v.eventSender = event.NewSender(v.Logger(), v.KeboolaProjectAPI())
+		v.eventSender = event.NewSender(v.Logger())
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/dependencies/mocked.go:147]: S posielanim eventov bolo viacero problemov.

Prvy commit je od @jprochazk , z https://github.com/keboola/keboola-as-code/pull/1207

^^^ tam sme neboli schopny doriesit testy a cakalo to na vylepsenie testov, rebasol som to do tohto PR a priadl nejake dalsie veci co som nasiel.

File: internal/pkg/service/buffer/event/sender.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (s *Sender) SendSliceUploadEvent(ctx context.Context, api *keboola.API, sta
 		}
 	}
 
-	s.sendEvent(ctx, api, start, err, "upload-slice", formatMsg, Params{
+	s.sendEvent(ctx, api, start, err, "slice-upload", formatMsg, Params{
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/event/sender.go:55]: Typo, druhy event sa vola `file-import`.

File: internal/pkg/service/buffer/event/sender.go [hunk 3]
```diff
@@ -164,10 +164,10 @@ func (s *Sender) sendEvent(ctx context.Context, api *keboola.API, start time.Tim
...
 			"lastRecordAt": params.Stats.LastRecordAt.String(),
 			"recordsCount": params.Stats.RecordsCount,
-			"recordsSize":  params.Stats.RecordsSize,
-			"bodySize":     params.Stats.BodySize,
-			"fileSize":     params.Stats.FileSize,
-			"fileGZipSize": params.Stats.FileGZipSize,
+			"recordsSize":  params.Stats.RecordsSize.Bytes(),
+			"bodySize":     params.Stats.BodySize.Bytes(),
+			"fileSize":     params.Stats.FileSize.Bytes(),
+			"fileGZipSize": params.Stats.FileGZipSize.Bytes(),
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/event/sender.go:170]: Pouzity `datasize` type je sice zalozeny na `uint64` ale serializuje sa do JSON ako `string`.

V [JSON scheme](https://github.com/keboola/event-schema/blob/130a890c67494dd9d22a769e5c35d9b86713010e/schema/ext.keboola.keboola-buffer..json#L33-L37) mame `int`.

Comment (michaljurecko) [reply internal/pkg/service/buffer/event/sender.go:170]: Samo o sebe to nepomohlo, kedze sme to posielali ako `form-data` v `go-client`.

Ale ked sa to preplo na `JSON body`, tak to uz ide, vid:
https://keboola.slack.com/archives/C02JAQXLY92/p1677853344346269

File: go.mod [hunk 4]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/jarcoal/httpmock v1.3.0
 	github.com/joho/godotenv v1.4.0
 	github.com/jpillora/longestcommon v0.0.0-20161227235612-adb9d91ee629
-	github.com/keboola/go-client v1.9.0
+	github.com/keboola/go-client v1.10.0
```
Comment (michaljurecko) [inline go.mod:31]: https://github.com/keboola/go-client/pull/81

**Comment by michaljurecko** [comment]
Test je doplneny v dalsom PR, kde sa riesia vylepsene worker E2E testy:
image

**Review by jprochazk — APPROVED** [review]

**Pull request merged**